src/client/app/index.ts

```diff
@@ -64,6 +65,9 @@ const VitePressApp = defineComponent({
 })
 
 export async function createApp() {
+  const target = getGlobalThis()
+  target.__VITEPRESS__ = true
```

Comment:
Try simply with `globalThis.__VITEPRESS__ = true` once. I don't think we need to polyfill it as we don't target very old browser.